js_modules/dagster-ui/packages/ui-core/src/instance/backfill/BackfillActionsMenu.tsx

```diff
@@ -0,0 +1,195 @@
+      backfill.numCancelable > 0)
+  );
+}
+
+export function backfillCanResume(backfill: {
+  hasResumePermission: boolean;
+  status: BulkActionStatus;
+  partitionSet: {__typename: 'PartitionSet'} | null;
+}) {
+  return (
+    backfill.hasResumePermission &&
+    backfill.status === BulkActionStatus.FAILED &&
+    backfill.partitionSet
+  );
+}
+
+export function backfillCanCancelRuns(
+  backfill: {hasCancelPermission: boolean},
+  counts: {[runStatus: string]: number} | null,
+) {
+  if (!backfill.hasCancelPermission || !counts) {
+    return false;
+  }
+  const queuedCount = counts[RunStatus.QUEUED] || 0;
+  const startedCount = counts[RunStatus.STARTED] || 0;
+  return queuedCount > 0 || startedCount > 0;
+}
+
+export const BackfillActionsMenu = ({
+  backfill,
+  counts,
+  refetch,
+}: {
+  backfill: BackfillActionsBackfillFragment;
+  counts: {[runStatus: string]: number} | null;
+  refetch: () => void;
+}) => {
+  const history = useHistory();
+  const runsUrl = runsPathWithFilters([
+    {
+      token: 'tag',
+      value: `dagster/backfill=${backfill.id}`,
+    },
+  ]);
+
+  const [showTerminateDialog, setShowTerminateDialog] = React.useState(false);
+  const [showStepStatus, setShowStepStatus] = React.useState(false);
+  const [resumeBackfill] = useMutation<ResumeBackfillMutation, ResumeBackfillMutationVariables>(
+    RESUME_BACKFILL_MUTATION,
+  );
+
+  const resume = async () => {
+    const {data} = await resumeBackfill({variables: {backfillId: backfill.id}});
+    if (data && data.resumePartitionBackfill.__typename === 'ResumeBackfillSuccess') {
+      refetch();
+    } else if (data && data.resumePartitionBackfill.__typename === 'UnauthorizedError') {
+      await showSharedToaster({
+        message: (
+          <Group direction="column" spacing={4}>
+            <div>
+              Attempted to retry the backfill in read-only mode. This backfill was not retried.
+            </div>
+          </Group>
+        ),
+        icon: 'error',
+        intent: 'danger',
+      });
+    } else if (data && data.resumePartitionBackfill.__typename === 'PythonError') {
+      const error = data.resumePartitionBackfill;
+      await showSharedToaster({
+        message: <div>An unexpected error occurred. This backfill was not retried.</div>,
+        icon: 'error',
+        intent: 'danger',
+        action: {
+          text: 'View error',
+          onClick: () =>
+            showCustomAlert({
+              body: <PythonErrorInfo error={error} />,
+            }),
+        },
+      });
+    }
+  };
+
+  return (
+    <>
+      <Popover
+        content={
+          <Menu>
+            {backfillCanCancelSubmission(backfill) ? (
+              <MenuItem
+                text="Cancel backfill submission"
+                icon="cancel"
+                intent="danger"
+                onClick={() => setShowTerminateDialog(true)}
+              />
```

Comment:
What do you think about making these visible but disabled (plus explanation tooltip), instead of hidden?

Reply:
Oh man good call... I think we should definitely do that instead 👀  In general having things disappear just makes it harder to learn the product I think

Reply:
Yeah this is much better:
<img width="554" alt="image" src="https://github.com/dagster-io/dagster/assets/1037212/db820d1a-1481-416a-92fd-0e74ca563330">